girder_worker/app.py

```diff
@@ -129,11 +136,57 @@ def describe(self):
     def call_item_task(self, inputs, outputs={}):
         return self.run.call_item_task(inputs, outputs)
 
+    def _maybe_transform_result(self, idx, result):
+        try:
+            grh = self.request.girder_result_hooks[idx]
+            if hasattr(grh, 'transform') and hasattr(grh.transform, '__call__'):
+                return grh.transform(result)
+        except IndexError:
+            return result
+
+    def _maybe_transform_argument(self, arg):
+        if hasattr(arg, 'transform') and hasattr(arg.transform, '__call__'):
+            return arg.transform()
```

Comment:
How does girder_client aka 'gc' [reference here]( https://github.com/girder/girder_worker_utils/blob/master/girder_worker_utils/transforms/girder_io.py#L12) get passed in, is that still todo?

Reply:
On the one hand you can use the transform like this:

```python
mytask.delay(GirderFileId('58dd37438d777f0aef5d8c75'))
```
This will create a default girder client with your permissions. Alternatively you
you can instantiate your own girder client with different permissions and pass it in using the ```gc``` keyword:

```python
token = Token().createToken(scope='some.custom.scope')
girder_client = GirderClient(apiUrl=some_api_url)
girder_client.token = token

mytask.delay(GirderFileId('58dd37438d777f0aef5d8c75', gc=girder_client))
```

This isn't currently documented anywhere,  but is the intended functionality

Reply:
Thanks for explanation.